File: LeetCode/11.盛水最多的容器/11.盛水最多的容器.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int low = 0;
        int high = height.size() - 1;
        int maxCap = 0;
        while(low < high){
            if (height[low] < height[high]) {
                maxCap = max(maxCap, height[low] * (high - low));
                low++;
            }else{
                maxCap = max(maxCap, height[high] * (high - low));
                high--;
            }
        } 
        return maxCap;
    }
};

File: LeetCode/11.盛水最多的容器/11.盛水最多的容器.go
func max(a int , b int) int {
    if a < b {
        return b
    }else {
        return a
    }
}
func maxArea(height []int) (maxCap int) {
    low := 0
    high := len(height) - 1
    maxCap = 0
    for{
        if(low >= high){
            break
        }
        if height[low] < height[high] {
           maxCap = max(maxCap, height[low] * (high - low))
           low++
        }else {
            maxCap = max(maxCap, height[high] * (high - low))
            high--
        }
    }
    return
}